src/test/scala/org/neo4j/spark/service/Neo4jQueryServiceTest.scala

```diff
@@ -138,7 +261,30 @@ class Neo4jQueryServiceTest {
 
     assertEquals("MATCH (source:`Person`) " +
       "MATCH (target:`Person`) " +
-      "MATCH (source)-[rel:`KNOWS`]->(target) WHERE (source.name = 'John Doe' OR target.name = 'John Doe') RETURN source, rel, target", query)
+      "MATCH (source)-[rel:`KNOWS`]->(target) WHERE (source.name = 'John Doe' OR target.name = 'John Doe') RETURN rel, source AS source, target AS target", query)
+  }
+
+  @Test
+  def testRelationshipAndFilterEqualTo(): Unit = {
+    val options: java.util.Map[String, String] = new java.util.HashMap[String, String]()
+    options.put(Neo4jOptions.URL, "bolt://localhost")
+    options.put("relationship", "KNOWS")
+    options.put("relationship.nodes.map", "true")
+    options.put("relationship.source.labels", "Person")
+    options.put("relationship.target.labels", "Person")
+    val neo4jOptions: Neo4jOptions = new Neo4jOptions(options)
+
+    val filters: Array[Filter] = Array[Filter](
+      EqualTo("source.id", "14"),
```

Comment:
can you change it to an int instead of a string? just to check if we handle the where correctly

Reply:
what do you mean? if we change to int here it will be an int in the query and result will be like 

`MATCH (source:`Person`) MATCH (target:`Person`) MATCH (source)-[rel:`KNOWS`]->(target) WHERE (source.id = 14 AND target.id = 16) RETURN rel, source AS source, target AS target`

Reply:
I can add a case in the IT for the DataSourceReader if there is none